apps/src/applab/Exporter.js

```diff
@@ -9,6 +9,136 @@ import elementLibrary from './designElements/library';
+  "levelGameName": true,
+  "skinId": true,
+  "baseUrl": true,
+  "app": true,
+  "droplet": true,
+  "pretty": true,
+  "level": {
+    "skin": true,
+    "editCode": true,
+    "embed": true,
+    "isK1": true,
+    "isProjectLevel": true,
+    "skipInstructionsPopup": true,
+    "disableParamEditing": true,
+    "disableVariableEditing": true,
+    "useModalFunctionEditor": true,
+    "useContractEditor": true,
+    "contractHighlight": true,
+    "contractCollapse": true,
+    "examplesHighlight": true,
+    "examplesCollapse": true,
+    "definitionHighlight": true,
+    "definitionCollapse": true,
+    "freePlay": true,
+    "appWidth": true,
+    "appHeight": true,
+    "sliderSpeed": true,
+    "calloutJson": true,
+    "disableExamples": true,
+    "showTurtleBeforeRun": true,
+    "autocompletePaletteApisOnly": true,
+    "textModeAtStart": true,
+    "designModeAtStart": true,
+    "hideDesignMode": true,
+    "beginnerMode": true,
+    "levelId": true,
+    "puzzle_number": true,
+    "stage_total": true,
+    "iframeEmbed": true,
+    "lastAttempt": true,
+    "submittable": true
+  },
+  "showUnusedBlocks": true,
+  "fullWidth": true,
+  "noHeader": true,
+  "noFooter": true,
+  "smallFooter": true,
+  "codeStudioLogo": true,
+  "hasI18n": true,
+  "callouts": true,
+  "channel": true,
+  "readonlyWorkspace": true,
+  "isLegacyShare": true,
+  "postMilestone": true,
+  "postFinalMilestone": true,
+  "puzzleRatingsUrl": true,
+  "authoredHintViewRequestsUrl": true,
+  "serverLevelId": true,
+  "gameDisplayName": true,
+  "publicCaching": true,
+  "is13Plus": true,
+  "hasContainedLevels": true,
+  "hideSource": true,
+  "share": true,
+  "labUserId": true,
+  "firebaseName": true,
+  "firebaseAuthToken": true,
+  "firebaseChannelIdSuffix": true,
+  "isSignedIn": true,
+  "pinWorkspaceToBottom": true,
+  "hasVerticalScrollbars": true,
+  "showExampleTestButtons": true,
+  "rackEnv": true,
+  "report": {
+    "fallback_response": true,
+    "callback": true,
+    "sublevelCallback": true,
+  },
+  "sendToPhone": true,
+  "send_to_phone_url": true,
+  "copyrightStrings": {
+    "thank_you": true,
+    "help_from_html": true,
+    "art_from_html": true,
+    "code_from_html": true,
+    "powered_by_aws": true,
+    "trademark": true,
+  },
+  "teacherMarkdown": true,
+  "dialog": {
+    "skipSound": true,
+    "preTitle": true,
+    "fallbackResponse": true,
+    "callback": true,
+    "sublevelCallback": true,
+    "app": true,
+    "level": true,
+    "shouldShowDialog": true,
+  },
+  "locale": true,
+};
+
+export function getAppOptionsFile() {
+  function getAppOptionsAtPath(whitelist, sourceOptions) {
+    if (!whitelist) {
+      return null;
+    }
+    const options = {};
+    Object.keys(whitelist).forEach(key => {
+      if (sourceOptions[key] === undefined) {
+        return;
+      }
+      if (typeof whitelist[key] === 'object') {
+        const subOptions = getAppOptionsAtPath(whitelist[key], sourceOptions[key]);
```

Comment:
I _think_ it's possible to pass `null` into the recursive call here, which doesn't get caught before a `sourceOptions[key]` call the next level down, which would cause a `TypeError`.